src/main/java/org/fenixedu/academic/ui/spring/AcademicBaseController.java

```diff
@@ -0,0 +1,39 @@
+package org.fenixedu.academic.ui.spring;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.ModelAttribute;
+
+public class AcademicBaseController {
+    //The HTTP Request that can be used internally in the controller
```

Comment:
Could you provide a reference to the Spring feature which covers this behavior?
I think we could refactor it to be done through dependency injection.
As long as I don't have to replicate code in different classes, any suggestion will be considered